plugins/modules/web_infrastructure/jira.py

```diff
@@ -81,18 +83,27 @@
     required: false
     description:
      - The comment text to add.
+     - Note that JIRA may not allow changing field values on specific transitions or states.
 
   status:
     type: str
     required: false
     description:
-     - The desired status; only relevant for the transition operation.
+     - Only used when C(operation) is I(transition), and a bit of aa misnomer, it actually refers to the transition name.
 
   assignee:
     type: str
     required: false
     description:
-     - Sets the assignee on create or transition operations. Note not all transitions will allow this.
+     - Sets the the assignee when C(operation) is I(create), I(transition) or I(edit).
+     - Recent versions of JIRA no longer accept an user name as an user identifier. In that case, use C(accountId) instead.
+     - Note that JIRA may not allow changing field values on specific transitions or states.
+
+  accountId:
```

Comment:
Ansible style would be:
```suggestion
  account_id:
```